### PR TITLE
WINDUP-835: Create an index/table of content as the default

### PR DESCRIPTION
**Pull request by mbriskar**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/640/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1267/

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java [hunk 1]
```diff
@@ -66,27 +65,13 @@
...
-
-    /**
-     * Indicates whether or not to display this in the list of all applications. Usually this would be true for a main "overview" type report for a
-     * particular application, and false for everything else.
-     */
-    @Property(DISPLAY_IN_APPLICATION_LIST)
-    void setDisplayInApplicationList(Boolean displayInApplicationList);
-
-    /**
-     * Indicates whether to display this report in the navigation index for the current application.
```
Comment (mbriskar) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java]: I think we have too much flags and this one basically maps the value sfrom Main_report flag.

Comment (jsight) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java]: +1

**Comment by mareknovotny** [comment]
It seems OK. We probably need a performance comparison for this change too as it could affect some big applications where Migration issues report will be generated twice in case of single application processing. Or am I wrong?

**Comment by mbriskar** [comment]
Yes, it is good to note that I had to change the number of generated migration_issues files for a single application to two. Not only it caused problems with linking before, but now when it was a main report for application, it was literally impossible to get to other application reports.

**Comment by jsight** [comment]
I'm -1 on any change that makes us produce the migration-issues report twice for a single analyzed app. There are other ways to fix that problem, IMO.

**Comment by mbriskar** [comment]
There are. It may not be generated at all.

**Comment by jsight** [comment]
I think that in the case of a single application, we should only have a single navigation bar. That bar should have links to all of the reports.

**Comment by mbriskar** [comment]
@jsight maybe redirecting?

**Comment by jsight** [comment]
@mbriskar I was just thinking that it doesn't need to generate a separate "global" navigation bar in that case.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1289/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/662/<h2>Build result: FAILURE</span></h2>[...truncated 48040 lines...][ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :windup-victims[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/pom.xml to org.jboss.windup.jenkins/windup-core-dependents-aggregate-pom/2.3.0-SNAPSHOT/windup-core-dependents-aggregate-pom-2.3.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/pom.xml to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT-sources.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT-sources.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/pom.xml to org.jboss.windup.quickstarts/windup-quickstarts-root/2.5.0-SNAPSHOT/windup-quickstarts-root-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.5.0-SNAPSHOT/windup-victims-2.5.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of 0e50b68b25be72e1b882b42d09fab715a8e2881a to FAILURE with url http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-core-PR-dependents-check/662/ and message: 'Build finished. 3 tests run, 0 skipped, 0 failed.'

**Comment by mbriskar** [comment]
Ok right now we generate migration issues only for multiple applications

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1290/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/663/<h2>Build result: FAILURE</span></h2>[...truncated 48040 lines...][ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :windup-victims[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/pom.xml to org.jboss.windup.jenkins/windup-core-dependents-aggregate-pom/2.3.0-SNAPSHOT/windup-core-dependents-aggregate-pom-2.3.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/pom.xml to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT-sources.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT-sources.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/pom.xml to org.jboss.windup.quickstarts/windup-quickstarts-root/2.5.0-SNAPSHOT/windup-quickstarts-root-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.5.0-SNAPSHOT/windup-victims-2.5.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of 98fd48c7980e673bedb8a454082905907b89e48c to FAILURE with url http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-core-PR-dependents-check/663/ and message: 'Build finished. 3 tests run, 0 skipped, 0 failed.'

**Comment by mareknovotny** [comment]
I can see the error in quickstart due the removal of org.jboss.windup.reporting.model.ApplicationReportModel.setDisplayInApplicationList(boolean)
http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-core-PR-dependents-check/663/org.jboss.windup.quickstarts$windup-victims/consoleText

**Comment by mareknovotny** [comment]
It seems fine to me now and if you sent PR for windup-quickstart I am +1 on this

**Comment by mbriskar** [comment]
https://github.com/windup/windup-quickstarts/pull/71

**Comment by jsight** [comment]
Merged http://github.com/windup/windup/compare/0592ec5...01c10c7

**Pull request closed without merging**